README.md

```diff
@@ -1,44 +1,235 @@
+```
+For more information on `type` syntax, see [Type Annotations](https://flow.org/en/docs/types/).
+
+**Data**
+
+Data is the result rows returned by the query as an array of maps.
+```
+[
+    {columnOne: 'one', columnTwo: 123},
+    {columnOne: 'two', columnTwo: 456}
+]
+```
+
+**Metadata**
+
+Metadata describes the shape of the results.
+```
+{
+    strings: ['columnOne'],
+    numbers: ['columnTwo'],
+    groups: []
+}
+```
+
+**Selection**
+
+`selection` is the current selection of the associated step as an Array of objects, with each object being one or more selected rows from the results.
+
+**setSelelction**
+
+`setSelection` is a callback passed in that allows the component to update the attached step's selection in Tableau CRM. In doing so, it potentially applies filters to the rest of the dashboard's contents depending on how the widgets are configured.
+```type SetSelection = (Selection) => void;```
+
+**Select Mode**
+
+Select mode describes which select mode the data source is in.
+
+**getState**
+
+`getState` is used to retrieve the current state of the dashboard. The state format is documented [here](https://help.salesforce.com/articleView?id=sf.bi_embed_filters.htm&type=5).
+
+**setState**
+
+`setState` is used to patch the current state of the dashboard.
+
+## Tableau CRM LWC Metadata
+
+Lightning Web Components used in Tableau CRM dashboards leverage the same file structure and [XML metadata format](https://developer.salesforce.com/docs/component-library/documentation/en/lwc/lwc.reference_configuration_tags) used by other LWCs on the Salesforce platform. Here are few additions to this metadata specific for Tableau CRM.
+
+**analytics__Dashboard Target**
+
+Adding this target to the list of `<targets>` for your component allows it to be used in Tableau CRM dashboards, provided that your component is public. You can also add a `<targetConfig>` for the new target to further customize how your component appears in dashboards.
+
+**`<hasStep>` Tag**
+
+In an `analyticsDashboard` target config, you can choose to include a `<hasStep>true</hasStep>` tag to indicate that your component requires an attached step to function as expected. With this tag, the dashboard builder UI prompts you to attach an existing step or create a new step when creating an instance of your component. Components with an attached step have access to step-specific properties like `results` and `selection`. For more information on these properties, see [Tableau CRM LWC Features](#tableau-crm-lwc-features).
+
+**Measure and Dimension Attribute Data Types**
+
+Attributes specified in a LWC's `analtyicsDashboard` target config are displayed in the Tableau CRM dashboard builder UI for configuration. In addition to the common data types, this target also supports `Measure` and `Dimension` data types for components with `<hasStep>true</hasStep>`. Dashboard builders are able to choose a column of the given data type from the results of the attached step.
+
```

Comment:
Would be nice to have a brief section here saying how to create your dashboard lwc with either the **SFDX: Create Analytics Dashboard LWC** command or via `sfdx force:lightning:component:create --type lwc -t [analyticsDashboard|analyticsDashboardWithStep] -n <name>` on the cli.